File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.py
def solution(n):
    # iteratable하게 만들기
    n_array = list(map(int, str(n)))
    # 내림차순 정렬
    n_array.sort(reverse=True)
    # 다시 하나의 정수로 합치기
    result = 0
    scale = 1
    for i in range(len(n_array)-1, -1, -1):
        result += scale*n_array[i]
        scale *= 10
    return result

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.*;
class Solution {
    public long solution(long n) {
        String str = String.valueOf(n);
        char[] arr = str.toCharArray();
        //System.out.println(arr);
        Arrays.sort(arr);
        str = new String(arr);
        str = new StringBuilder(str).reverse().toString();
        return Long.parseLong(str);
    }
}